python_modules/dagster/dagster/_core/definitions/asset_reconciliation_sensor.py

```diff
@@ -48,6 +50,96 @@
     from dagster._utils.caching_instance_queryer import CachingInstanceQueryer  # expensive import
 
 
+class _AutoMaterializeCondition(Enum):
+    FRESHNESS = "FRESHNESS"
+    DOWNSTREAM_FRESHNESS = "DOWNSTREAM_FRESHNESS"
+    PARENT_UPDATED = "PARENT_UPDATED"
+    MISSING = "MISSING"
+
+
+@whitelist_for_serdes
+class AutoMaterializeReason(NamedTuple):
+    condition: _AutoMaterializeCondition
+
+    @staticmethod
+    def freshness() -> "AutoMaterializeReason":
+        return AutoMaterializeReason(condition=_AutoMaterializeCondition.FRESHNESS)
+
+    @staticmethod
+    def downstream_freshness() -> "AutoMaterializeReason":
+        return AutoMaterializeReason(condition=_AutoMaterializeCondition.DOWNSTREAM_FRESHNESS)
+
+    @staticmethod
+    def parent_updated() -> "AutoMaterializeReason":
+        return AutoMaterializeReason(condition=_AutoMaterializeCondition.PARENT_UPDATED)
+
+    @staticmethod
+    def missing() -> "AutoMaterializeReason":
+        return AutoMaterializeReason(condition=_AutoMaterializeCondition.MISSING)
+
+
+class _AutoMaterializeSkipCondition(Enum):
+    PARENT_UNRECONCILED = "PARENT_UNRECONCILED"
+
+
+@whitelist_for_serdes
+class AutoMaterializeSkipReason(NamedTuple):
+    condition: _AutoMaterializeSkipCondition
+
+    @staticmethod
+    def parent_unreconciled() -> "AutoMaterializeSkipReason":
+        return AutoMaterializeSkipReason(
+            condition=_AutoMaterializeSkipCondition.PARENT_UNRECONCILED
+        )
+
+
+@whitelist_for_serdes
+class AutoMaterializeDecision(NamedTuple):
```

Comment:
nit: AutoMaterializeDecision vs AutoMaterializeEvaluation? I'm fine with either, so far @braunjj has been using evaluation in designs